### ISSUE TITLE
shell.trashItem() fails on tmpfs filesystems

### ISSUE THREAD

**Issue by ckerr** [electron/electron#28045]
tldr: there are edge cases where `gio trash` will fail and in those cases we could use a fallback approach to trashing files in `shell/common/platform_util_linux.cc`'s `PlatformTrashItem()`.

Same issue exists in the older blocking-API function `shell.moveItemToTrash()`.

The same `gio trash` trash can be seen from the command line. For example:

```sh
$ df --output=fstype,target /home
Type  Mounted on
ext4 /home
$ echo 'Hello, World!' > "$HOME"/test.txt
$ gio trash "$HOME"/test.txt
$ echo $?
0
$
$ df --output=fstype,target /tmp
Type Mounted on
tmpfs /tmp
$ echo 'Hello, World!' > /tmp/test.txt
$ gio trash /tmp/test.txt
$ gio: file:///tmp/test.txt: Trashing on system internal mounts is not supported
$ echo $? 
1
```

Downstream related Atom issue: https://github.com/atom/tree-view/issues/1237
Upstream related glib issue: https://gitlab.gnome.org/GNOME/glib/-/issues/1722

### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * 11.0.0, 12.0.0
* **Operating System:**
  * Linux
* **Last Known Working Electron version:**
  * None

### Expected Behavior

The file should be trashed

### Actual Behavior

The file is not trashed (though it does correctly throw an exception that the operation failed)

### To Reproduce

https://gist.github.com/7fd1ca22aa15b089fc6573a0dd67841d
